app/utils/expectations.js

```diff
@@ -125,10 +136,14 @@ const conditionalAlternativeId = 'uses_conditional_alternative'
 const conditionalRepetitionId = 'uses_conditional_repetition'
 const simpleRepetitionId = 'uses_simple_repetition'
 const declaresProcedureId = 'declares_procedure'
+const doesNotNestControlStructuresId = 'does_not_nest_control_structures'
 
 export const isCritical = (expectationResult) => expectationResult && expectationResult.isCritical
 
+export const notCritical = (expectationResult) => !isCritical(expectationResult) && !warningInControlStructureBlock(expectationResult)
+
+export const warningInControlStructureBlock = (expectationResult) => expectationResult && expectationResult.warningInControlStructureBlock
```

Comment:
¿Por qué no optional chaining?

```js
(expectationResult) => expectationResult?.warningInControlStructureBlock
```

Y en realidad, si vas a hacer typecheck, mejor no setear otro atributo en el objeto result, y hacer:

```suggestion
export const warningInControlStructureBlock = (expectationResult) => expectationResult.id == doesNotNestControlStructuresId
```

Idem abajo para `isUsageResult`. Igual sobre isUsage voy a tirar un par de issues.